File: src/main/java/org/academy/api/client/hud/terminal/TerminalConfig.java
package org.academy.api.client.hud.terminal;

import org.academy.api.client.config.KeyBindingConfig;
import org.academy.api.common.gson.TypeHandler;

public final class TerminalConfig extends KeyBindingConfig {
    public static final class Action implements TypeHandler<TerminalConfig> {
        public static final TypeHandler<TerminalConfig> INSTANCE = new Action();

        private Action() {
        }

        @Override
        public TerminalConfig getDefault() {
            return new TerminalConfig();
        }

        @Override
        public Class<TerminalConfig> getTypeClass() {
            return TerminalConfig.class;
        }
    }
}

File: src/main/kotlin/org/academy/api/client/hud/terminal/TerminalConfig.kt
package org.academy.api.client.hud.terminal

import org.academy.api.client.config.KeyBindingConfig
import org.academy.api.common.gson.TypeHandler

class TerminalConfig : KeyBindingConfig() {
    var blurRadius: Float = 20f

    class Action private constructor() : TypeHandler<TerminalConfig> {
        override fun getDefault(): TerminalConfig {
            return TerminalConfig()
        }

        override fun getTypeClass(): Class<TerminalConfig> {
            return TerminalConfig::class.java
        }

        companion object {
            val INSTANCE: TypeHandler<TerminalConfig> = Action()
        }
    }
}
